### PR TITLE
cdc: return sink congested error explictly

### LINKED ISSUE #17503: cdc should report congested region error
When send events meet memory quota exceed, cdc report `region_not_found` error to the ticdc, this is confusing. Instead, make cdc report the region congested should be better.

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What is changed and how it works?

Issue Number: Close #17503

image

What's Changed:

```commit-message
TiKV-CDC report `Congested` error if allocate memory quota failed meet `MemoryQuotaExceed`
* Convert the `SendError::Congested` to `cdcpb::Error::Congested`
```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: components/cdc/src/errors.rs [hunk 1]
```diff
@@ -99,18 +99,27 @@ impl Error {
...
-        } else {
-            // TODO: Add more errors to the cdc protocol
-            let mut region_not_found = errorpb::RegionNotFound::default();
-            region_not_found.set_region_id(region_id);
-            err_event.set_region_not_found(region_not_found);
+        match self {
+            Error::Sink(SendError::Congested) => {
+                let mut congested = cdcpb::Congested::default();
+                congested.set_region_id(region_id);
+                err_event.set_congested(congested);
```
Comment (overvenus) [inline components/cdc/src/errors.rs:106]: How does old TiCDC handle this error? Will it see an empty error message?

Comment (3AceShowHand) [reply components/cdc/src/errors.rs:106]: The old TiCDC treat this `congested` error as unknown error, print logs.

File: components/cdc/src/errors.rs [hunk 2]
```diff
@@ -99,18 +99,27 @@ impl Error {
...
+            }
+            _ => {
```
Comment (hicqu) [inline components/cdc/src/errors.rs:108]: It's better to use real pattern instead of `_`.

**Review by overvenus — APPROVED** [review]

**Review by hicqu — APPROVED** [review]

**Pull request merged**